tasks/node/agent/version_test.go

```diff
@@ -15,132 +16,130 @@ func TestNodeAgentVersion(t *testing.T) {
+				expectedReturn := ""
+				Expect(output).To(Equal(expectedReturn))
 			})
 		})
 
 	})
 
-	Describe("getNodeVerFromLog()", func() {
+	Describe("Execute()", func() {
 		var (
-			incomingLogs []logtask.LogElement
-			output       []logNodeAgentVersion
+			options  tasks.Options
+			upstream map[string]tasks.Result
+			result   tasks.Result
 		)
 
 		JustBeforeEach(func() {
-			output = getNodeVerFromLog(incomingLogs)
+			result = p.Execute(options, upstream)
 		})
 
-		Context("When given a logElement containing a Node Agent version", func() {
+		Context("When given a Node Agent Module isn't found", func() {
 			BeforeEach(func() {
-				incomingLogs = []logtask.LogElement{
-					logtask.LogElement{
-						FileName: "newrelic_agent.log",
-						FilePath: "../../fixtures/node/",
-					},
-				}
-			})
-			It("Should return the expected agent version, log file location, and true", func() {
-				expectedReturn := []logNodeAgentVersion{
-					logNodeAgentVersion{
-						Logfile:      "../../fixtures/node/newrelic_agent.log",
-						AgentVersion: "1.38.2",
-						MatchFound:   true,
+				upstream = map[string]tasks.Result{
+					"Node/Env/Dependencies": tasks.Result{
+						Status: tasks.Info,
+						Payload: []NodeEnv.NodeModuleVersion{
+							{
+								Module:  "Apollo",
+								Version: "1.2.3",
+							},
+							{
+								Module:  "lodash",
+								Version: "4.2.03",
+							},
+						},
 					},
-				}
-				Expect(output).To(Equal(expectedReturn))
-			})
-		})
-
-		Context("When given a logElement which does not contain a Node Agent Version", func() {
-			BeforeEach(func() {
-				incomingLogs = []logtask.LogElement{
-					logtask.LogElement{
-						FileName: "newrelic-python-agent.log",
-						FilePath: "../../fixtures/python/root/tmp/",
+					"Node/Config/Agent": tasks.Result{
+						Status: tasks.Success,
 					},
 				}
 			})
-			It("Should return an empty Node agent version, the log file location, and false", func() {
-				expectedReturn := []logNodeAgentVersion{
-					logNodeAgentVersion{
-						Logfile:      "../../fixtures/python/root/tmp/newrelic-python-agent.log",
-						AgentVersion: "",
-						MatchFound:   false,
-					},
+			It("Should return a tasks.Warning", func() {
+				expectedReturn := tasks.Result{
+					Status:  tasks.Warning,
+					Summary: "We were unable to find the 'newrelic' module required for the Node Agent installation. Make sure to run 'npm install newrelic' and verify that 'newrelic' is listed in your package.json.",
 				}
-				Expect(output).To(Equal(expectedReturn))
+				Expect(result).To(Equal(expectedReturn))
 			})
 		})
-
-		Context("When given a logElement with a malformed filepath", func() {
+		Context("When given a Node Agent Module is found", func() {
 			BeforeEach(func() {
-				incomingLogs = []logtask.LogElement{
-					logtask.LogElement{
-						FileName: "newrelic_agent.BADFILE.log",
-						FilePath: "../../fixtures/node/",
+				upstream = map[string]tasks.Result{
```

Comment:
I added one additional test for a successful case when we do get a node agent version